applications/vanilla/models/class.discussionmodel.php

```diff
@@ -1167,6 +1159,67 @@ public function calculate(&$discussion) {
         $this->fireEvent('SetCalculatedFields');
     }
 
+    /**
+     * Calculate if the user has read all the Comments in a discussion. The data in the Discussion table
+     * is more likely to be reliable because when Comments are deleted we update the Discussion table
+     * but not the UserDiscussion table.
+     *
+     * @param int $discussionCommentCount Number of Comments according to the Discussion table.
+     * @param mixed $discussionLastCommentDate Date of last Comment according to the Discussion table.
+     * @param int $userReadComments Number of Comments the user has read according to the UserDiscussion table.
+     * @param mixed $userLastReadDate Date of last Comment read according to the UserDiscussion table.
+     * @return array
```

Comment:
Given that this returns an array with a specific format, you should document the return.